Remove Simulation Bounds from Simulation Templates

Mythicaeda opened the pull request:
* **Tickets addressed:** Hotfix
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
<!-- What approach was taken to satisfy the ticket being addressed? What should reviewers be aware of? -->
- Removes the ability for a simulation template to specify the simulation bounds.
- Forces the sim bounds in the sim config to be non null. 
- Initializes the sim bounds to the plan bounds on creation of a simulation row
- Enforces the sim end bound to be >= sim start bound. The "or equal to" was added so that plans with duration zero can be created, although I'm not sure what their use case would be.
- Removed Optional wrapper around return of `GetSimulationTemplateAction.get()`

## Verification
<!-- How were the changes validated? Were any automated tests added, updated, removed, or re-baselined? -->
- Updated Temporal Subset Simulation e2eTests and removed those that no longer apply (ie those that tested whether bounds specified in the sim config or sim config template took precedence)
- Updated "Verify posting of simulation results" test in scheduler

## Documentation
<!-- What documentation was invalidated by these changes? Which artifacts should reviewers check for accuracy and completeness? -->
- [ ] Docs issue [here](https://github.com/NASA-AMMOS/aerie-docs/issues/32).

## Future work
<!-- What next steps can we anticipate from here, if any? -->

Mythicaeda committed:
Make simulation_start/end_time non nullable

- Remove columns from simulation_template

Mythicaeda committed:
Remove SimulationStart- and EndTime from SimulationTemplateRecord

mattdailis commented on e2e-tests/src/tests/scheduler.test.ts:
```diff
@@ -272,9 +270,9 @@ test.describe('Scheduling', () => {
-          producerIsThere = true
-          expect(topic.events.length).toEqual(0)
+        default:
+          test.fail(topic.events.length !== 0, "Unexpected topic with events: "+topic.name)
```
Is this change simply refactoring, or is it related to the changes to simulation templates made in this PR? If it's simply refactoring, could we pull it into its own commit?

Mythicaeda replied:
Done 👍

mattdailis commented on merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulationTemplateAction.java:
```diff
@@ -28,13 +27,11 @@ public GetSimulationTemplateAction(final Connection connection) throws SQLExcept
   }
 
-  public Optional<SimulationTemplateRecord> get(final long simulationTemplateId)
+  public SimulationTemplateRecord get(final long simulationTemplateId)
```
If it's no longer safe to call this action on a non-existent simulation template id - can we add a javadoc comment documenting what it will do if it is provided a non-existent id? It is true that it is currently called in a safe fashion, but by changing the behavior from "returns Optional.empty()" to "throws an exception, probably?", we've made it more difficult for a future programmer to use this action safely.

Mythicaeda replied:
Done 👍. Also added a similar documentation for GetSimulationAction, where the same change was made in a prior PR.

Adjacently, I'm not sure if this method is now more "unsafe". Obviously it throws an exception rather than returning an empty Optional if there are no results, but blindly using that Optional without checking if the value is present would already lead to an exception (with a slightly obfuscated root cause, depending on how far from the `get` the value is blindly used), and I'm not certain the safety differences between checking the Optional as soon as it's returned and throwing an exception if it's empty and throwing an exception in the method itself. Could you please clarify that part?

Mythicaeda replied:
Resolving as I've dropped the commit this conversation is about.

mattdailis reviewed (commented):
SQL changes look good to me - just a couple comments about the e2e tests and the GetSimulationTemplate action

Mythicaeda committed:
Fix Temporal Subset e2e tests

Mythicaeda committed:
Update scheduler e2eTest

- Test "Verify posting of simulation results" now has a default case that captures all topics without any events

mattdailis reviewed (approved):
Looks good to me 👌
